docs/reference/sso.md

```diff
@@ -0,0 +1,522 @@
+
+Steps:
+1. In the Azure portal, in App registrations, select your application.
+2. Under Manage, select Authentication.
+3. Under Platform configurations, select Add a platform.
+4. Under Configure platforms, select the web option.
+5. Select Configure to complete the platform configuration.
+
+Once you finish registering lakeFS Cloud with Azure AD, save the **Application (Client) ID**, **Application Secret Value** and send this to Treeverse's team to finish the integration.
+
+## Enterprise
+
+lakeFS Enterprise provides a secondary service, running side-by-side with lakeFS which handles the authentication, this service is called Fluffy.
+Once you've onboarded to [lakeFS Enterprise](../enterprise.md) and have access to the authentication service (Fluffy), you can configure one of the supported authentication methods below.
+lakeFS Enterprise leverage Helm's capabilities to run the sidecar with it's configuration, on top of the specific-authentication method, please review the [Helm configuration](#helm).
+
+### AD FS (using SAML)
+{: .d-inline-block }
+Enterprise
+{: .label .label-purple }
+
+> **Note**: AD FS integration is using certificates to sign & encrypt requests going out from Fluffy and decrypt incoming requests from AD FS server.
+
+In order for fluffy to work, some values must be configured, update the relevant values section in this file and comment it out.
+1. Replace `fluffy.saml_rsa_public_cert` and `fluffy.saml_rsa_private_key` with real certificate values
+2. Replace `fluffyConfig.auth.saml.idp_metadata_url` to the metadata URL of the provider <adfs-auth.company.com>
+3. Replace `fluffyConfig.auth.saml.external_user_id_claim_name` with the claim name representing user id name in ADFS
+4. Replace `lakefs.company.com` with the lakeFS server endpoint.
+
+If you'd like to generate the certificates using OpenSSL, you can take a look at the following example:
+```sh
+openssl req -x509 -newkey rsa:2048 -keyout myservice.key -out myservice.cert -days 365 -nodes -subj "/CN=lakefs.company.com" -
+```
+
+lakeFS Enterprise Configuration:
+```yaml
+lakefsConfig: |
+  # Important: make sure to include the rest of your lakeFS Configuration here!
+
+  auth:
+    cookie_auth_verification:
+      auth_source: saml
+      friendly_name_claim_name: displayName
+      external_user_id_claim_name: samName
+      default_initial_groups:
+        - "Developers"
+    logout_redirect_url: "https://lakefs.company.com/logout-saml"
+    encrypt:
+      secret_key: shared-secrey-key
+    ui_config:
+      login_url: "https://lakefs.company.com/sso/login-saml"
+      logout_url: "https://lakefs.company.com/sso/logout-saml"
+      login_cookie_names:
+        - internal_auth_session
+        - saml_auth_session
```

Comment:
is there a better way to articulate how to deal with this bleeding chunk of configuration? for example, "Replace the existing `auth` section of your lakeFS server config with the following" ?